### PR TITLE
Feature/stc eddy heat fluxes

### PR DESCRIPTION
**Pull request by zdlawrence**
**Description**
This PR is intended to add the first stratosphere-troposphere coupling POD to the MDTF-diagnostics package. 
This POD focuses on assessing the statistical relationships of vertical wave activity measured by eddy heat fluxes 
on the stratosphere and troposphere. 

**How Has This Been Tested?**
This POD was tested on the NOAA PSL Linux computer systems using the MTDF_python3_base conda environment.
I used CMIP6 data downloaded directly from the freely available [Google Cloud](https://console.cloud.google.com/marketplace/details/noaa-public/cmip6?_ga=2.256360716.-68223858.1655222241&pli=1) archive to develop and test this POD. 
The POD takes about 2-4 minutes to run when using 30-50 years of monthly frequency data, though 
much of this time is spent by MDTF to read and subset the fields. 

**Checklist:**
- [x] I have reviewed my own code to ensure that if follows the [POD development guidelines](https://mdtf-diagnostics.readthedocs.io/en/latest/sphinx/dev_guidelines.html)
- [x] My branch is up-to-date with the NOAA-GFDL develop branch, and all merge conflicts are resolved
- [x] The scripts are written in Python 3.7 or above (preferred; required if funded by a CPO grant), NCL, or R
- [x] All of my scripts are in the diagnostics/[POD short name] subdirectory, and include a main_driver script, template html, and settings.jsonc file
- [x] I have made corresponding changes to the documentation in the POD's doc/ subdirectory
- [x] If applicable, I've added a .yml file to src/conda, and my environment builds with `conda_env_setup.sh` 
- [x] I have added any necessary data to input_data/obs_data/[pod short name] and/or input_data/model/[pod short name]
- [x] My code is portable; it uses [MDTF environment variables](https://mdtf-diagnostics.readthedocs.io/en/latest/sphinx/ref_envvars.html), and does not contain hard-coded file or directory paths
- [x] I have provided the code to generate digested data files from raw data files
- [x] Each digested data file generated by the script contains numerical data (no figures), and is 3 GB or less in size
- [x] The repository contains no extra test scripts or data files

### CODE REVIEW

**Review by wrongkindofdoctor — COMMENTED** [review]
@zdlawrence Nice work! I don't see any major issues, save for the possible removal of some hardcoded paths noted in the following comment. Note that I changed the target branch to main, since we just officially confirmed that the develop branch will be deprecated and removed.

File: diagnostics/stc_eddy_heat_fluxes/make_era5-rean_pod-data.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+# air temperature
+# geopotential (this is post-processed to geopot height by dividing by 9.80665)
+
+import numpy as np
+import xarray as xr
+
+# These variables are contained in individual files, across all months
+vwnd = xr.open_dataset('/Projects/era5/Monthlies/pressure/vwnd.mon.mean.nc')
+air = xr.open_dataset('/Projects/era5/Monthlies/pressure/air.mon.mean.nc')
+hgt = xr.open_dataset('/Projects/era5/Monthlies/pressure/hgt.mon.mean.nc')
```
[the remark this answers is not in the document]
Comment (zdlawrence) [reply diagnostics/stc_eddy_heat_fluxes/make_era5-rean_pod-data.py]: Good point, thanks! I've updated these to use strings set by generic environment variables.

File: diagnostics/stc_eddy_heat_fluxes/doc/stc_eddy_heat_fluxes.rst [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  temperatures
+- Lag-correlation plots of polar cap geopotential heights at different pressure
+  levels and months with early-season eddy heat fluxes at 100 hPa
+
+These plots are made for both hemispheres; for the Northern Hemisphere (NH),
+they focus on the early (DJF or Dec) and late winter (JFM). For the Southern
+Hemisphere (SH), they focus on the early (ASO or Sep) and late spring (SON).
+These months are when coupling between the stratosphere and troposphere are
+most active in the respective hemispheres. 
+
```
[the remark this answers is not in the document]
Comment (zdlawrence) [reply diagnostics/stc_eddy_heat_fluxes/doc/stc_eddy_heat_fluxes.rst:61]: Sorry for the delay on this. I've added some additional discussion about the weather/climate implications related to this POD, as per your recommendation.

**Review by wrongkindofdoctor — APPROVED** [review]

File: diagnostics/stc_eddy_heat_fluxes/doc/stc_eddy_heat_fluxes.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (wrongkindofdoctor) [reply diagnostics/stc_eddy_heat_fluxes/doc/stc_eddy_heat_fluxes.rst:61]: @zdlawrence Thank you for the updates! I emailed John to review your changes.

[the remark this answers is not in the document]
Comment (jkrasting) [reply diagnostics/stc_eddy_heat_fluxes/doc/stc_eddy_heat_fluxes.rst:61]: Much improved, thanks!

**Review by jkrasting — APPROVED** [review]
Looks good to me.  I flagged one spot in `src_eddy_heat_fluxes.py` where a custom font is specified.  As long as this does not pose an issue if the font is missing, this PR looks good to me.  Thanks for the added documentation!

**Comment by wrongkindofdoctor** [comment]
@zdlawrence The STC Eddy Heat Fluxes POD is now merged into the main repo. I think I still have the test dataset when I was helping debug the code a while back, so you should be all set. Please note that developers and end-users may reach out for assistance with POD-specific issues in the future.

**Pull request merged**